### PR TITLE
Types: some minor improvements to `ta`

### PR DESCRIPTION
**Pull request by AlexKornitzer**
Not all pointers to structs start at the beginning, thanks Microsoft for your implementation of list entries (LIST_ENTRY). Thus I have tweaked `ta` so that it can now take an additional argument `offset`. So for example we can now do the following: `ta _LDR_DATA_TABLE_ENTRY.DllBase 0x10`, as we are pointing not at the start of the structure but actually at offset 0x10 (the InInitializationOrderLinks field).
I have also extended the r_type_get_struct_memb function so that it will now parse inline structs rather than incorrectly bailing out. So below we can see that BaseDllName (UNICODE_STRING) is correctly resolved to the inlined structure.

Example of new improvements:
image

NOTE: Annoyingly cmd_type.c is formatted badly so there are lots of additions due to linting!

### CODE REVIEW

File: libr/core/cmd_type.c [hunk 1]
```diff
@@ -41,7 +41,7 @@ static const char *help_msg_t_minus[] = {
 static const char *help_msg_ta[] = {
 	"Usage: ta[...]", "", "",
 	"tas", " <offset>", "List all matching structure offsets",
-	"ta", " <struct member>", "Change immediate to structure offset",
+	"ta", " <struct member> [offset]", "Change immediate to structure offset",
```
Comment (radare) [inline libr/core/cmd_type.c]: isnt this [struct.member] ?

Comment (AlexKornitzer) [reply libr/core/cmd_type.c]: Probably but did not want to change more than necessary

**Comment by AlexKornitzer** [comment]
Have I handled adding optional args correctly, I just copied it from elsewhere but it is done differently in different commands!

**Comment by radare** [comment]
no i mean the help message was wrong

**Comment by AlexKornitzer** [comment]
Do you want me to change it to the suggestion above?

File: libr/core/cmd_type.c [hunk 2]
```diff
@@ -1017,7 +1016,13 @@ static int cmd_type(void *data, const char *input) {
 			free (off);
 		} break;
 		case ' ': {
-			char *type = strdup (input + 2);
+			const char *off = NULL;
```
Comment (AlexKornitzer) [inline libr/core/cmd_type.c]: Nah I meant, is this the correct way to handle optional arguments?

File: libr/core/cmd_type.c [hunk 3]
```diff
@@ -1017,7 +1016,13 @@ static int cmd_type(void *data, const char *input) {
...
+			char *space, *type = strdup (r_str_trim_ro (input + 2));
+			space = strchr (type, ' ');
```
Comment (radare) [inline libr/core/cmd_type.c]: define and assign in the same line. its easier to read

**Pull request merged**